ui/components/multichain/edit-accounts-modal/edit-accounts-modal.test.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { EthAccountType, KeyringAccountType } from '@metamask/keyring-api';
+import { renderWithProvider } from '../../../../test/jest/rendering';
+import mockState from '../../../../test/data/mock-state.json';
+import configureStore from '../../../store/store';
+import { EditAccountsModal } from '.';
+
+const render = (
+  props: {
+    onClose: () => void;
+    allowedAccountTypes: KeyringAccountType[];
+  } = {
+    onClose: () => jest.fn(),
+    allowedAccountTypes: [EthAccountType.Eoa, EthAccountType.Erc4337],
+  },
+  state = {},
+) => {
+  const store = configureStore({
+    ...mockState,
+    metamask: {
+      ...mockState.metamask,
+      ...state,
+      permissionHistory: {
+        'https://test.dapp': {
+          eth_accounts: {
+            accounts: {
+              '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': 1709225290848,
+            },
+          },
+        },
+      },
+    },
+    activeTab: {
+      origin: 'https://test.dapp',
+    },
+  });
+  return renderWithProvider(<EditAccountsModal {...props} />, store);
+};
+describe('EditAccountsModal', () => {
+  it('should render correctly', () => {
```

Comment:
can we add more unit test for this component:
- calls onClose when the close button is clicked
- handles checkbox click

Reply:
+1. Also if possible, a test for the Select All button

Reply:
Checkbox click is for account list item component and it's included there and so is the onClose for modal but I added a test for select all